### ISSUE TITLE
Sails Models should support instanceof

### ISSUE THREAD

**Issue by wgreenberg** [balderdashy/sails#2359]
Currently there seems to be no way of verifying the class of a given waterline record. Suppose we have a model called User and we want to check that a given record is an instance of User:

```
User.create().then(function(newUser) {
    console.log(newUser instanceof User.constructor) // false
});
```

As far as I can tell, the only waterline-y way of doing this (as opposed to a method which uses Sails routes for model identification) is a hack from this StackOverflow question: http://stackoverflow.com/questions/26446206/how-to-get-the-model-name-or-model-class-of-a-waterline-record/26447442#26447442

Models in Sails should at least support calls to `instanceof` like their Waterline model counterparts do

**Comment by adambiggs** [comment]
+1

Any updates on this?

**Comment by sailsbot** [comment]
Thanks for posting, @wgreenberg.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by kevinburkeshyp** [comment]
As far as I can tell this is still a defect

**Comment by vajogaspar** [comment]
In sails@v0.11 it worked like this: 
```
myuser instanceof User._model.__bindData__[0] === true
```
however, in sails@v0.12 it's still ok when running in the interactive console, but not ok during the execution of bootstrap.js. I tried to use it after the orm hook loaded, but still no luck.
I try to use the sails models with cancan.js - any thoughts/suggestions?
Thanks a lot!

**Comment by cyrilchapon** [comment]
Any solution | workaround | update on this... ? Trying to put some unit tests together and I didn't find anything on this.

**Comment by huafu** [comment]
So far the only workaround is in that SO answer http://stackoverflow.com/questions/26446206/how-to-get-the-model-name-or-model-class-of-a-waterline-record/26447442#26447442

**Comment by cyrilchapon** [comment]
Yes I saw, but it [doesn't work in sails 0.12](http://stackoverflow.com/questions/26446206/how-to-get-the-model-name-or-model-class-of-a-waterline-record/26447442#comment60515805_26447442) =(

**Comment by huafu** [comment]
@cyrilchapon I used it in sails `0.12` and not only in the console. Watch out, the code you posted isn't the same as the one on the linked SO answer.
